projects/igniteui-angular/src/lib/services/excel/excel-files.ts

```diff
@@ -404,7 +406,8 @@ export class WorkbookFile implements IExcelFile {
  */
 export class ContentTypesFile implements IExcelFile {
     public writeElement(folder: JSZip, worksheetData: WorksheetData) {
-        folder.file('[Content_Types].xml', ExcelStrings.getContentTypesXML(!worksheetData.isEmpty, worksheetData.options.exportAsTable));
+        const hasSharedStrings = !worksheetData.isEmpty || (worksheetData.isEmpty && worksheetData.options.alwaysExportHeaders);
```

Comment:
Do we need the "(worksheetData.isEmpty &&" portion? If alwaysExportHeaders is true, we will have shared strings with or without data, right?

Reply:
yes but we shouldn't have shared strings if alwaysExportHeaders is false and data is empty

Reply:
The result from !worksheetData.isEmpty || (worksheetData.isEmpty && worksheetData.options.alwaysExportHeaders) is the same as !worksheetData.isEmpty || worksheetData.options.alwaysExportHeaders.